src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -252,6 +252,41 @@ public static IEnumerable<IEdmVocabularyAnnotation> FindVocabularyAnnotationsInc
             return result;
         }
 
+        /// <summary>
+        /// Gets a target path's vocabulary annotations defined in a specific model and models referenced by that model.
+        /// </summary>
+        /// <param name="model">The model to search.</param>
+        /// <param name="targetPath">The target path to check for annotations.</param>
+        /// <returns>Annotations attached to the target path by this model or by models referenced by this model.</returns>
+        public static IEnumerable<IEdmVocabularyAnnotation> FindVocabularyAnnotations(this IEdmModel model, string targetPath)
+        {
+            EdmUtil.CheckArgumentNull(model, nameof(model));
+            EdmUtil.CheckArgumentNull(targetPath, nameof(targetPath));
+
+            IEdmTargetPath edmTargetPath = model.GetTargetPath(targetPath);

```

Comment:
>IEdmTargetPath edmTargetPath = model.GetTargetPath(targetPath); [](http://example.com/codeflow?start=12&length=63)

***Importantly again:

Since you call "GetTargetPath(...)" to get the IEdmTargetPath instance, be sure it's not same object stored in the model. So, when we use the new edmTargetPath to call 'FindVocabularyAnnotations", make sure we are retrieve the same

Reply:
@xuzhg are you proposing that `model.GetTargetPath` should return the same `IEdmTargetPath` instance for the same the same string path?

I see the problem you're alluding to. It it returns a different instance for the same target path instance, then we'll have duplicate entries added to the cache, which can lead to a memory leak.

Reply:
@KenitoInc Assuming the following is scenario where the value for `Target` is the same for the two `Annotations` element is valid, does `model.GetTargetPath` work as expected?

```xml
<Annotations Target="NS.Default/Customers/Address/NS.DerivedAddress">
    <Annotation Term="Org.OData.Capabilities.V1.SortRestrictions">
        <Record>
            <PropertyValue Property="Sortable" Bool="true" />
            <PropertyValue Property="AscendingOnlyProperties">
                <Collection />
            </PropertyValue>
            <PropertyValue Property="DescendingOnlyProperties">
                <Collection />
            </PropertyValue>
            <PropertyValue Property="NonSortableProperties">
                <Collection>
                    <PropertyPath>MyRoad</PropertyPath>
                </Collection>
            </PropertyValue>
        </Record>
    </Annotation>
</Annotations>
<Annotations Target="NS.Default/Customers/Address/NS.DerivedAddress">
    <Annotation Term="Org.OData.Capabilities.V1.FilterRestrictions">
        <Record>
            <PropertyValue Property="Filterable" Bool="true" />
            <PropertyValue Property="RequiresFilter" Bool="true" />
            <PropertyValue Property="RequiredProperties">
                <Collection />
            </PropertyValue>
            <PropertyValue Property="NonFilterableProperties">
                <Collection>
                    <PropertyPath>MyRoad</PropertyPath>
                </Collection>
            </PropertyValue>
        </Record>
    </Annotation>
</Annotations>
```

Reply:
+1

Reply:
Yes, That's my concern.

Reply:
@gathogojr @ElizabethOkerio @xuzhg 
This is the second or 3rd time I'm sharing this.
Check this test `ParsingMultipleOutOfLineAnnotationsWorks` in the `CsdlReaderTests.TargetPath.cs`.

Technically in the example above, we have 2 annotations with 2 different `terms` but the same `EdmTargetPath`.

`IEdmTargetPath targetPath = model.GetTargetPath("NS.Default/Customers/Address/NS.DerivedAddress");` will return the correct `EdmTargetPath`.

`IEnumerable<IEdmVocabularyAnnotation> targetPathStringAnnotations = model.GetTargetPathAnnotations("NS.Default/Customers/Address/NS.DerivedAddress");` will return the 2 annotations.

Reply:
Is this supposed to be written like this? 

<Annotations Target="NS.Default/Customers/Address/NS.DerivedAddress">
    <Annotation Term="Org.OData.Capabilities.V1.SortRestrictions">
        <Record>
            <PropertyValue Property="Sortable" Bool="true" />
            <PropertyValue Property="AscendingOnlyProperties">
                <Collection />
            </PropertyValue>
            <PropertyValue Property="DescendingOnlyProperties">
                <Collection />
            </PropertyValue>
            <PropertyValue Property="NonSortableProperties">
                <Collection>
                    <PropertyPath>MyRoad</PropertyPath>
                </Collection>
            </PropertyValue>
        </Record>
    </Annotation>
    <Annotation Term="Org.OData.Capabilities.V1.FilterRestrictions">
        <Record>
            <PropertyValue Property="Filterable" Bool="true" />
            <PropertyValue Property="RequiresFilter" Bool="true" />
            <PropertyValue Property="RequiredProperties">
                <Collection />
            </PropertyValue>
            <PropertyValue Property="NonFilterableProperties">
                <Collection>
                    <PropertyPath>MyRoad</PropertyPath>
                </Collection>
            </PropertyValue>
        </Record>
    </Annotation>
</Annotations>